### PR TITLE
Add prefix to output file for barrnap

### PR DESCRIPTION
**Pull request by sofstam**
## PR checklist

Updates barrnap module to include prefix in the output filename.

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Comment by charles-plessy** [comment]
Hi! As you changed the name of the files produced by the module, the regression tests fail because they were not updated to the new names.  In principle https://github.com/nf-core/modules/blob/master/tests/modules/nf-core/barrnap/test.yml should be updated, but I think that nf-core team members would love if the module were converted to the use of nf-test instead… https://nf-co.re/docs/tutorials/tests_and_test_data/nf-test_writing_tests

**Comment by sofstam** [comment]
Thank you, will update to nf-test.

File: modules/nf-core/barrnap/main.nf [hunk 1]
```diff
@@ -32,7 +32,7 @@ process BARRNAP {
         --threads $task.cpus \\
         --kingdom $db \\
         $input \\
-        > rrna_${db}.gff
+        > ${prefix}_rrna_${db}.gff
```
Comment (SPPearce) [inline modules/nf-core/barrnap/main.nf:35]: Can we remove the `rrna` here?

File: modules/nf-core/barrnap/meta.yml
Comment (SPPearce) [inline modules/nf-core/barrnap/meta.yml:1]: Can you update the input here, to match what is being used in the actual script.

File: modules/nf-core/barrnap/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    tag "modules"
+    tag "modules_nfcore"
+    tag "barrnap"
+
+    test("barrnap") {
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
```
Comment (SPPearce) [inline modules/nf-core/barrnap/tests/main.nf.test:15]:
```suggestion
```

File: modules/nf-core/barrnap/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    test("barrnap - stub") {
+
+        options "-stub"
+
+        when {
+            params {
+                outdir = "$outputDir"
+            }
```
Comment (SPPearce) [inline modules/nf-core/barrnap/tests/main.nf.test:41]:
```suggestion
```

File: modules/nf-core/barrnap/main.nf
Comment (SPPearce) [inline modules/nf-core/barrnap/main.nf:1]: The stub output naming doesn't match the main script.

Comment (sofstam) [reply modules/nf-core/barrnap/main.nf:1]: Updated.

File: modules/nf-core/barrnap/main.nf (hunk 1 above)
Comment (sofstam) [reply modules/nf-core/barrnap/main.nf]: Done

File: modules/nf-core/barrnap/main.nf [hunk 4]
```diff
@@ -46,7 +46,7 @@ process BARRNAP {
     def prefix = task.ext.prefix ?: "${meta.id}"
 
     """
-    touch ${prefix}.gff
+    touch ${prefix}_${db}.gff
```
Comment (SPPearce) [inline modules/nf-core/barrnap/main.nf:49]: Does this see the  `db         = dbname ? "${dbname}" : 'bac'` defined above? I suspect you'll need this here too.

Comment (sofstam) [reply modules/nf-core/barrnap/main.nf:49]: Updated the PR.

**Review by SPPearce — APPROVED** [review]
Ok, good to go now.

**Pull request merged**